lib/src/rules/unnecessary_statement.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.unnecessary_statement;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc =
+    r"Don't write unnecessary statements neither use getters with side-effect.";
+
+const _details = r'''
+
+**DON'T** use unnecessary statements neither use getters with side-effect.
+
+**BAD:**
+```
+void main() {
+  int a;
+  a; // LINT
+}
+```
+
+**GOOD:**
+```
+void main() {
+  int a;
+  foo(a);
+}
+```
+
+''';
+
+class UnnecessaryStatement extends LintRule {
+  _Visitor _visitor;
+  UnnecessaryStatement()
+      : super(
+            name: 'unnecessary_statement',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _UnnecessaryStatementVisitor extends UnifyingAstVisitor {
+  var isUnnecessaryStatement = true;
+
+  @override
+  visitAssignmentExpression(AssignmentExpression node) {
+    isUnnecessaryStatement = false;
+  }
+
+  @override
+  visitBinaryExpression(BinaryExpression node) {
+    if (!isUnnecessaryStatement) {
+      return;
+    }
+    if (node.operator.type == TokenType.EQ_EQ ||
+        node.operator.type == TokenType.BANG_EQ) {
```

Comment:
Yes, but it is not invoked when you compare with null right?, so this part check if it is something like a == null, if it is that, then == it is not invoked and it does not have side-effects